chpl/chpl-service/src/main/java/gov/healthit/chpl/aspect/QuestionableActivityListener.java

```diff
@@ -58,16 +49,70 @@ public void setEnvironment(final Environment e) {
 
     /**
      * Any activity added with a reason would be handled here.
-     * @param concept - ActivityConcept
-     * @param objectId - Long
-     * @param activityDescription - String
-     * @param originalData - Object
-     * @param newData - Object
+     * @param concept
+     * @param objectId
+     * @param activityDescription
+     * @param originalData
+     * @param newData
+     * @param reason
      */
+    @After("execution(* gov.healthit.chpl.manager.impl.ActivityManagerImpl.addActivity(..)) && "
+            + "args(concept,objectId,activityDescription,originalData,newData,reason,..)")
+    public void checkQuestionableActivityWithReason(final ActivityConcept concept,
+            final Long objectId, String activityDescription, final Object originalData,
+            final Object newData, final String reason) {
+        LOGGER.info("Called QuestionableActivityAspect2.checkQuestionableActivityWithReason()");
+        if (originalData == null || newData == null
+                || !originalData.getClass().equals(newData.getClass())
+                || Util.getCurrentUser() == null) {
+            return;
+        }
+
+        //all questionable activity from this action should have the exact same date and user id
+        Date activityDate = new Date();
+        Long activityUser = Util.getCurrentUser().getId();
+
+        if (originalData instanceof CertifiedProductSearchDetails
+                && newData instanceof CertifiedProductSearchDetails) {
+            CertifiedProductSearchDetails origListing = (CertifiedProductSearchDetails) originalData;
+            CertifiedProductSearchDetails newListing = (CertifiedProductSearchDetails) newData;
+
+            //look for any of the listing questionable activity
+            questionableActivityManager.checkListingQuestionableActivity(origListing, newListing, activityDate, activityUser, reason);
```

Comment:
Does this check also check against the "grace period window"? Or does the grace period check need to be before this call?